parsers/ES.py

```diff
@@ -127,139 +146,110 @@ def fetch_island_data(
 
 @refetch_frequency(timedelta(days=1))
 def fetch_consumption(
-    zone_key: ZONE_KEYS,
+    zone_key: ZoneKey,
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> List[dict]:
+    check_valid_parameters(zone_key, session, target_datetime, logger)
+
     ses = session or Session()
     island_data = fetch_island_data(zone_key, ses, target_datetime)
-    data = []
-    if island_data:
-        for response in island_data:
-            response_data = {
-                "zoneKey": zone_key,
-                "datetime": get(response.timestamp).datetime,
-                "consumption": response.demand,
-                "source": "demanda.ree.es",
-            }
-
-            data.append(response_data)
-
-        return data
-    else:
-        raise ParserException(
-            "ES.py",
-            f"No consumption data returned for zone: {zone_key}",
-            zone_key,
+    consumption = TotalConsumptionList(logger)
+    for event in island_data:
+        consumption.append(
+            zoneKey=zone_key,
+            datetime=datetime.utcfromtimestamp(event.timestamp).astimezone(
+                timezone.utc
```

Comment:
Double checking, is the island data in UTC as well or local time ?

Reply:
As far as I know the whole demanda.ree.es platform use UTC timestamps (not dates) but I can double check this.